04.ls/ls.rb

```diff
@@ -33,4 +33,119 @@ def split_files(files)
   divided_files
 end
 
+HALF_YEAR = 15_552_000
+
+FILE_TYPE = {
+  'fifo' => 'p',
+  'characterSpecial' => 'c',
+  'directory' => 'd',
+  'blockSpecial' => 'b',
+  'file' => '-',
+  'link' => 'l',
+  'socket' => 's'
+}.freeze
+
+PERMISSION = {
+  '0' => '---',
+  '1' => '--x',
+  '2' => '-w-',
+  '3' => '-wx',
+  '4' => 'r--',
+  '5' => 'r-x',
+  '6' => 'rw-',
+  '7' => 'rwx'
+}.freeze
+
+def show_long_format(path, file_names)
+  absolute_path_list = file_names.map { |file_name| File.join(path, file_name) }
+  stats_by_path = absolute_path_list.to_h { |absolute_path| [absolute_path, File.lstat(absolute_path)] }
```

Comment:
```suggestion
  stats_by_path = file_names.to_h { |file_name| ... }
```

全ファイルのループはなるべく減らした方が効率が良くなります。

Reply:
ループを1回にまとめました。
```ruby
stats_by_path = file_names.to_h do |file_name|
    absolute_path = File.join(path, file_name)
    [absolute_path, File.lstat(absolute_path)]
end
```